chromium_src/chrome/browser/policy/profile_policy_connector.cc

```diff
@@ -3,22 +3,25 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
+#include "chrome/browser/browser_process.h"
 #include "components/policy/core/common/configuration_policy_provider.h"
+#include "components/prefs/pref_service.h"
 
 namespace brave_policy {
 std::unique_ptr<policy::ConfigurationPolicyProvider>
-CreateBraveProfilePolicyProvider();
+CreateBraveProfilePolicyProvider(PrefService* local_state);
 }  // namespace brave_policy
 
 // Create and Init the Brave Profile Policy Provider which is used
 // for Brave Origin and other Brave specific policies.
 // We do not need to define a patch for Shutdown since anything in
 // wrapped_policy_providers_ will automatically call Shutdown.
-#define BRAVE_PROFILE_POLICY_CONNECTOR_INIT                         \
-  auto provider = brave_policy::CreateBraveProfilePolicyProvider(); \
-  brave_profile_policy_provider_ = provider.get();                  \
-  policy_providers_.push_back(provider.get());                      \
-  provider->Init(schema_registry);                                  \
+#define BRAVE_PROFILE_POLICY_CONNECTOR_INIT                       \
```

Comment:
I think you can get away with no changes to chromium_src overrides. I.e. you don't need to have `local_state` passed into the profile policy provider.